integr8ly/integreatly-operator
# MGDAPI-1170: Improve product operator installation

# Description

### Duplicated install plans

Currently, as part of the product operators installation, the retrieval of the InstallPlans associated with the product is done by
listing all InstallPlans in the Subscription namespace. OLM however might create multiple InstallPlans where only one is valid and is referenced in the Subscription status as an `InstallPlanRef` field. The current approach will loop through all the InstallPlans in the namespace, approve them an wait for the status to be `Complete`, but if an invalid InstallPlan comes first in the list, it will remain stuck in progress, never reaching the valid one.

Change the `MarketPlaceInterface` and its `Manager` implementation to retrieve only the InstallPlan that is referenced by the Subscription, and update the unit tests that depend on this method.

### Missing deployment in CSVs

It's been noted that CSVs have been created with missing deployments, signaling the product operator as installed but the operator doesn't start running. Add logic to validate the CSVs and retry the installation if the CSV is found to be invalid

## Verification

1. Install a cluster with Openshift ~4.6.8~
2. Install RHOAM from this PR
    > While RHOAM progresses the products installation, check the InstallPlans in the product operator namespaces, check if there are more than one InstallPlan for namespace, this might happen arbitrarily
3. Verify that the installation completes successfilly

## Type of change




## Checklist
